### PR TITLE
Implements Hokuto Shinken

### PR DESCRIPTION
**Pull request by Ezuo**
Idle hands are the devil's playthings, and so I have gifted 2D spessmen with this boon. Now you too can tell people that they are already dead. 

For the uninitiated, Hokuto Shinken is the martial art used in Fist of the North Star, most famously known by the line "Omae Wa Mou Shindeiru". Upon the request of Garnascus BSTs will spawn with it.

The movelist is as follows
>Hakkei no Ho: Passive. Deflects all projectiles.
>Sokin Jizai Kyaku: Harm Harm Disarm. Paralyses an opponent for several seconds.
>Goshi Retsu Dan: Disarm Disarm Harm Disarm. Explodes an opponent's hand.
>Gekitsui Shi: Harm Disarm Harm Disarm Harm. They are already dead.

### CODE REVIEW

**Comment by Pacmandevil** [comment]
heavy rp

**Comment by garnascus** [comment]
i will always support silly/meme stuff like this to adminbus in.

**Comment by ParadoxSpace** [comment]
muh hyakuretsu ken

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+/datum/martial_art/hokuto_shinken
+	name = "Hokuto Shinken"
+	deflection_chance = 100
+	help_verb = /datum/martial_art/hokuto_shinken/proc/hokuto_shinken_help
+
+
+/datum/martial_art/hokuto_shinken/proc/check_streak(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(findtext(streak,SOKIN_JIZAI_KYAKU_COMBO))
+		streak = ""
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: Why are you emptying streak here?

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		Sokin_Jizai_Kyaku(A,D)
+		return 1
+	if(findtext(streak,GOSHI_RETSU_COMBO))
+		streak = ""
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: and here.

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		Goshi_Retsu(A,D)
+		return 1
+	if(findtext(streak,GEKITSUI_SHI_COMBO))
+		streak = ""
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: and here.

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	else if(A.zone_sel.selecting == "l_hand" || A.zone_sel.selecting == "r_hand")
+		A.visible_message("<span class='warning'>[A] strikes [D]'s [organ.name] with two extended fingers!</span>")
+		admin_attack_log(A, D, "exploded [organ.joint].", "had his [organ.joint] exploded.", "exploded [organ.joint] of")
+		playsound(get_turf(A), "strike_hit", 50, 1, -1)
+		organ.droplimb(0, DROPLIMB_BLUNT)
+		return 1
+	else
+		playsound(get_turf(A), "punch", 50, 1, -1)
+		D.apply_damage(15, BRUTE)
+		A.visible_message("<span class='warning'>[A] strikes [D] with their closed fist!</span>")
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: Shouldn't this be `a closed fist`?

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		return 1
+	return 1
+
+/datum/martial_art/hokuto_shinken/proc/Gekitsui_Shi(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	log_and_message_admins("[A] gibbed [D] with Hokuto Shinken.")
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: LAMA prepends user automatically. Change this to:
`log_and_message_admins("gibbed [D] with Hokuto Shinken.", A)`

File: code/modules/martial_arts/hokuto_shinken.dm [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+	A.do_attack_animation(D)
+	A.visible_message("<span class='warning'>[A] strikes [D]\'s forehead with four extended fingers!</span>")
+	playsound(get_turf(A), "sound/voice/already_dead.ogg", 50)
+	D.Stun(10)
+	A.say("Omae Wa Mou Shindeiru.")
+	addtimer(CALLBACK(D, /mob/.proc/say, "Nani?!"), 25)
+	addtimer(CALLBACK(D, /mob/.proc/gib), 33)
+	addtimer(CALLBACK(GLOBAL_PROC, /proc/playsound, D.loc, "sound/magic/Disintegrate.ogg", 50, 1, -1), 33)
```
Comment (Lohikar) [inline code/modules/martial_arts/hokuto_shinken.dm]: this and the above should probably be combined into a proc, though this is fine as is.

File: code/modules/martial_arts/hokuto_shinken.dm (hunk 5 above)
Comment (Lohikar) [reply code/modules/martial_arts/hokuto_shinken.dm]: Probably should be log attack, or change `D` to `key_name(D)`.

File: code/modules/martial_arts/hokuto_shinken.dm (hunk 1 above)
Comment (Ezuo) [reply code/modules/martial_arts/hokuto_shinken.dm]: That is just how the martial art moves are designed, the streak is always emptied when one procs. I assume that is to stop spammable moves, or to keep the streak clear after doing a move.

File: code/modules/martial_arts/hokuto_shinken.dm (hunk 4 above)
Comment (Ezuo) [reply code/modules/martial_arts/hokuto_shinken.dm]: The grammar checks out to me.

**Comment by LordFowl** [comment]
Is martial arts just going to be the new low-effort memecode dump now?

**Comment by Arrow768** [comment]
I agree with LordFowl on this.

What´s the point of it if its just given to BSTs ?
And if its not just given to BSTs, than there are severe balance issues.

File: code/modules/martial_arts/hokuto_shinken.dm (hunk 4 above)
Comment (Lohikar) [reply code/modules/martial_arts/hokuto_shinken.dm]: The grammar implies they only have one fist.

Comment (Arrow768) [reply code/modules/martial_arts/hokuto_shinken.dm]: No, it implies that they strike them with *one* firsts.

"their closed fists" would mean they hit them with both of their fists.

**Comment by Alberyk** [comment]
Okay, I am against this because:
-it is just a reference that is not really used anywhere
-it is a reference that already exist in the code, with the plasma fist martial art
-why are you even giving it to the bluespace tech that is supossed to be a testing mob not a joke
-we should find ways to balance/incorporate martial into the game, mostly as an antag tool

**Comment by Arrow768** [comment]
No response from the author to the concerns raised.
Closing it.

**Pull request closed without merging**